src/Application.cxx

```diff
@@ -171,79 +163,21 @@ int main(int argc, char** argv) {
 
     auto runManager = new G4RunManager;
 
-    auto det = new DetectorConstruction();
-
-    runManager->SetUserInitialization(det);
-    runManager->SetUserInitialization(new PhysicsList(restPhysList));
-
-    auto prim = new PrimaryGeneratorAction(det);
-
-    if (restG4Metadata->GetParticleSource(0)->GetEnergyDistType() == "TH1D") {
-        TString fileFullPath = (TString)restG4Metadata->GetParticleSource(0)->GetSpectrumFilename();
-
-        TFile fin(fileFullPath);
-
-        TString sptName = restG4Metadata->GetParticleSource(0)->GetSpectrumName();
+    auto detector = new DetectorConstruction(fSimulationManager);
 
-        TH1D* h = (TH1D*)fin.Get(sptName);
-
-        if (!h) {
-            cout << "REST ERROR  when trying to find energy spectrum" << endl;
-            cout << "File : " << fileFullPath << endl;
-            cout << "Spectrum name : " << sptName << endl;
-            exit(1);
-        }
-
-        initialEnergySpectrum = *h;
-
-        Double_t minEnergy = restG4Metadata->GetParticleSource(0)->GetMinEnergy();
-        if (minEnergy < 0) minEnergy = 0;
-
-        Double_t maxEnergy = restG4Metadata->GetParticleSource(0)->GetMaxEnergy();
-        if (maxEnergy < 0) maxEnergy = 0;
-
-        // We set the initial spectrum energy provided from TH1D
-        prim->SetSpectrum(&initialEnergySpectrum, minEnergy, maxEnergy);
-    }
+    runManager->SetUserInitialization(detector);
+    runManager->SetUserInitialization(new PhysicsList(fSimulationManager->fRestGeant4PhysicsLists));
 
-    if (restG4Metadata->GetParticleSource(0)->GetAngularDistType() == "TH1D") {
-        TString fileFullPath = (TString)restG4Metadata->GetParticleSource(0)->GetAngularFilename();
+    runManager->SetUserInitialization(new ActionInitialization(fSimulationManager));
 
-        TFile fin(fileFullPath);
-
-        TString sptName = restG4Metadata->GetParticleSource(0)->GetAngularName();
-        TH1D* h = (TH1D*)fin.Get(sptName);
-
-        if (!h) {
-            cout << "REST ERROR  when trying to find angular spectrum" << endl;
-            cout << "File : " << fileFullPath << endl;
-            cout << "Spectrum name : " << sptName << endl;
-            exit(1);
-        }
-
-        initialAngularDistribution = *h;
-
-        // We set the initial angular distribution provided from TH1D
-        prim->SetAngularDistribution(&initialAngularDistribution);
-    }
-
-    auto run = new RunAction(prim);
-
-    auto event = new EventAction();
-
-    auto track = new TrackingAction(run, event);
-    auto step = new SteppingAction();
-
-    runManager->SetUserAction(run);
-    runManager->SetUserAction(prim);
-    runManager->SetUserAction(event);
-    runManager->SetUserAction(track);
-    runManager->SetUserAction(step);
+    auto step = (SteppingAction*)G4RunManager::GetRunManager()->GetUserSteppingAction();
+    auto primaryGenerator =
+        (PrimaryGeneratorAction*)G4RunManager::GetRunManager()->GetUserPrimaryGeneratorAction();
```

Comment:
```suggestion
    auto step = (SteppingAction*)G4RunManager::GetRunManager()->GetUserSteppingAction();

    auto primaryGenerator =
(PrimaryGeneratorAction*)G4RunManager::GetRunManager()->GetUserPrimaryGeneratorAction();
```

Reply:
Whats the difference? I think this is the way the .clang-format outputs it right? maybe there is an ambiguity in these cases

Reply:
Well, just put a line break between step and primary generator, I think it is more readable this way. I have issues with the github editor.